core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/BaseAwsJsonProtocolFactory.java

```diff
@@ -252,6 +269,7 @@ public final SubclassT registerModeledException(ExceptionMetadata errorMetadata)
          * @param exceptionBuilderSupplier Suppplier of the base service exceptions Builder.
          * @return This builder for method chaining.
          */
+        @Deprecated
         public final SubclassT defaultServiceExceptionSupplier(Supplier<SdkPojo> exceptionBuilderSupplier) {
```

Comment:
can we call out in javadoc its deprecated in favor of new createErrorResponseHandler overload, for future reference

Reply:
+1, yes will do!

Reply:
I removed the deprecation for this and deprecated `BaseAseJsonProtocolFactory#createErrorResponseHandler(JsonOperationMetadata)` and `registerModeledException` instead. I'll add a backlog item to remove our usages of these two methods in favor of the new ``BaseAseJsonProtocolFactory#createErrorResponseHandler(JsonOperationMetadata, Function)` introduced here.

Reply:
Backlog item created, ref JAVA-7976.